lib/foreman/renderer/scope/macros/helpers.rb

```diff
@@ -11,6 +11,36 @@ module Helpers
             sections only: %w[all]
           end
 
+          apipie :method, "Returns counter increased by given value (as String) stored in a hostgroup param named 'sequence_num_xxx'" do
+            required :hostgroup_name, String, "hostgroup title to store in (e.g. RHEL7/base)"
+            optional :increment, Integer, "integer increment, can be negative too", default: 1
+            optional :name, String, "name of the sequence (prefix for hostgroup parameter)", default: 'default'
+            optional :prefix, Integer, "leading zeroes", default: 1
+            returns String, "String with incremented value or empty string if hostgroup not found"
+            example 'sequence_hostgroup_param_next("myHG", 10) #=> 10'
+            example 'sequence_hostgroup_param_next("myHG", 2, "kiosk", 5) #=> 00002'
+          end
+          def sequence_hostgroup_param_next(hostgroup_name, increment = 1, name = "default", prefix = 1)
+            # This helper cannot be in a host-context because it was built mainly for discovery naming
+            # where host-hostgroup was not yet set. This macro allows accessing hostgroup parameters
+            # named starting with "sequence_num_" and increasing those counters which is not a security
+            # threat.
+            hostgroup = Hostgroup.unscoped.find_by_title(hostgroup_name)
```

Comment:
`unscoped`? What context this is usually rendered in? Can I touch the counter of a hostgroup I don't have access to? Or is this rendered under internal admin always? If that's the case, we don't need `unscoped` at all, since admin always has access everywhere.

Reply:
My initial thinking was there cannot be any harm, the first version only allowed to increase the counter. But now it's possible to decrease too. I will remove unscoped, yeah.